src/main/java/auth/service/UserServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package auth.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import auth.dao.UserRepository;
+import auth.model.User;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Override
+	public void save(User user) {
+		user.setPassword(passwordEncoder.encode(user.getPassword()));
```

Comment:
I think it will be better to use a listener on User entity to listen for beforeSave event. It can potentially reduce the risk of accidentally injecting userRepository bean to save user object instead of userService.